jobs/gtfs-rt-parser/gtfs_rt_parser.py

```diff
@@ -123,109 +218,440 @@ def identify_files(glob, rt_file_type: RTFileType, progress=False) -> List[RTFil
+        if verbose:
+            log(
+                f"no schedule data found for {hour.source_files[0].schedule_path}",
+                fg=typer.colors.YELLOW,
+                pbar=pbar,
+            )
+        errors.append(
+            RTFileProcessingError(
+                step="download_schedule",
+                exception=str(e),
+                hive_path=hour.data_hive_path,
+            )
+        )
+    except subprocess.CalledProcessError as e:
+        msg = f"WARNING: execute_rt_validator failed for {dst_path_rt} and {gtfs_zip}"
+        if verbose:
+            msg += f"\nincluded files: {included_files}\n{e.stderr}"
+        log(
+            msg,
+            fg=typer.colors.RED,
+            pbar=pbar,
+        )
+        errors.append(
+            RTFileProcessingError(
+                step="validate",
+                exception=str(e),
+                hive_path=hour.data_hive_path,
+                body=e.stderr,
+            )
+        )
+
+    for rt_file in hour.source_files:
+        try:
+            with open(
+                os.path.join(
+                    dst_path_rt,
+                    rt_file.timestamped_filename + ".results.json",
+                )
+            ) as f:
+                records = json.load(f)
+        except FileNotFoundError:
+            if verbose:
+                log(
+                    f"WARNING: no validation output found in {str(rt_file.timestamped_filename)}",
+                    fg=typer.colors.YELLOW,
+                    pbar=pbar,
+                )
+            continue
+
+        for record in records:
+            records_to_upload.append(
+                {
+                    # back and forth so we can use pydantic serialization
+                    "metadata": json.loads(rt_file.json()),
+                    **record,
+                }
+            )
 
-        if not parsed or "entity" not in parsed:
-            msg = f"WARNING: no records found in {str(rt_file.path)}"
-            if pbar:
-                pbar.write(msg)
-                pbar.update(1)
-            else:
-                typer.secho(msg, fg=typer.colors.YELLOW)
-            return
+    if records_to_upload:
+        upload_gzipped_records(
+            fs,
+            gzip_fname=str(
+                tmp_dir + f"/validation_{hour.suffix}" + JSONL_GZIP_EXTENSION
+            ),
+            out_path=f"{hour.validation_hive_path}{JSONL_GZIP_EXTENSION}",
+            records=records_to_upload,
+            pbar=pbar,
+            verbose=verbose,
+        )
+    else:
+        log(
+            f"WARNING: no records at all for {hour.validation_hive_path}",
+            fg=typer.colors.YELLOW,
+            pbar=pbar,
+        )
+
+    if errors:
+        upload_gzipped_records(
+            fs,
+            gzip_fname=str(tmp_dir + f"/errors_{hour.suffix}" + JSONL_GZIP_EXTENSION),
+            out_path=f"{hour.errors_hive_path}_validation{JSONL_GZIP_EXTENSION}",
```

Comment:
**question**: Why do we affix "_validation" here if it's also present in the second level folder name? Just for extra clarity?

Reply:
Because the parse errors and validation errors are written out separately, we'd overwrite one or the other without the extra prefix.

I agree it's unclear, I will make it another hive path attribute.

Reply:
Ahhh yeah I misunderstood this: https://github.com/cal-itp/data-infra/blob/aab55d004aad4a70d55215aae525c58ccf2c1db3/jobs/gtfs-rt-parser/gtfs_rt_parser.py#L134 and thought that "file type" included the validation or not distinction. I think that this should be specified at the top level underneath the bucket -- these errors should end up in separate tables (parse vs. validate errors).